R/derive_vars_trt_start.R

```diff
@@ -0,0 +1,158 @@
+#' Derive Datetime of First Exposure to Treatment
+#'
+#' Derives datetime of first exposure to treatment (`TRTSDTM`)
+#'
+#' @param dataset Input dataset
+#'
+#'   The variables specified by the `subject_keys` parameter are expected.
+#'
+#' @param dataset_ex `ex` dataset
+#'
+#'   The variables `EXSTDTC`, `EXSEQ`, and those specified by the `filter_ex`
+#'   `dtc` and `subject_keys`parameters are expected.
+#'
+#' @param filter_ex Filter condition for the ex dataset
+#'
+#'   Only observations of the ex dataset which fulfill the specified condition
+#'   are considered for the treatment start date.
+#'
+#'   Default: NULL
+#'
+#'   Permitted Values: logical expression
+#'
+#' @param subject_keys Variables to uniquely identify a subject
+#'
+#' A list of quosures where the expressions are symbols as returned by
+#' `vars()` is expected.
+#'
+#' @param new_var name of treatment start date e.g. `TRTSDTM`
+#'
+#' @param dtc name of variable from which start date is derived e.g. `EXSTDTC`
+#'
+#' @param date_imputation The value to impute the day/month when a datepart is
+#'   missing.
+#'
+#'   If `NULL`: no date imputation is performed and partial dates are returned as
+#'   missing.
+#'
+#'   Otherwise, a character value is expected, either as a
+#'   - format with month and day specified as 'mm-dd': e.g. `"06-15"` for the 15th
+#'   of June,
+#'   - or as a keyword: `"FIRST"`, `"MID"`, `"LAST"` to impute to the first/mid/last
+#'   day/month.
+#'
+#'   Default is `NULL`.
+#'
+#' @param time_imputation The value to impute the time when a timepart is
+#'   missing.
+#'
+#'   A character value is expected, either as a
+#'   - format with hour, min and sec specified as 'hh:mm:ss': e.g. `"00:00:00"`
+#'   for the start of the day,
+#'   - or as a keyword: `"FIRST"`,`"LAST"` to impute to the start/end of a day.
+#'
+#'   Default is `"00:00:00"`.
+#'
+#' @param min_dates Minimum dates
+#'  A list of dates is expected. It is ensured that the imputed date is not
+#'  before any of the specified dates, e.g., that the imputed adverse event start
+#'  date is not before the first treatment date. Only dates which are in the
+#'  range of possible dates of the dtc value are considered. The possible dates
+#'  are defined by the missing parts of the dtc date (see example below). This
+#'  ensures that the non-missing parts of the dtc date are not changed.
+#'
+#' @param max_dates Maximum dates
+#'
+#' A list of dates is expected. It is ensured that the imputed date is not after
+#' any of the specified dates, e.g., that the imputed date is not after the data
+#' cut off date. Only dates which are in the range of possible dates are
+#' considered.
+#'
+#' @param ord_vars
+#'  Order of variables for which either the last or first observation
+#'  is picked as treatment start date
+#'  e.g. vars(EXSTDTC,EXSEQ)
+#'
+#' @param ord_filter
```

Comment:
I agree that for treatment start date we need `mode = "first"` only.

However, at the moment the code of both functions is the same. Thus I wonder if we need _two_ functions. We could keep the `mode` parameter and just implement one function.

What do you think?

Reply:
I thought about that when I was writing this, if we go with one function what should be the name? then put in documentation that start/end is controlled by the mode parameter?

Reply:
Maybe `derive_trt_dt()`?

Reply:
What about :`derive_vars_trt_date()` and 2 wrapper functions :   `derive_vars_trt_start` with mode="first", and `derive_vars_trt_end` would mode="last"?

Reply:
Maybe we should drop the "trt" part because the function is no longer specific for treatment dates. The only reference to treatment is the parameter name `dataset_ex` but the function would work with any dataset.

What about naming the function `derive_vars_extreme_date()` and renaming `dataset_ex` to `dataset_date`?

Reply:
In favor 👍